src/pynxtools_xps/kratos/metadata_kratos.py

```diff
@@ -0,0 +1,350 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=too-many-lines,too-many-instance-attributes
+
+import re
+import datetime
+from typing import Any, Dict, List, Union
+from pathlib import Path
+import pytz
+
+from pynxtools_xps.reader_utils import (
+    convert_pascal_to_snake,
+)
+
+from pynxtools_xps.value_mappers import (
+    convert_bool,
+    convert_units,
+)
+
+from pynxtools_xps.kratos.kratos_data_model import (
+    KratosMetadata,
+)
+
+SETTINGS_MAP: Dict[str, str] = {
+    "location_i_d": "location_id",
+    "tilt": "sample_tilt",
+    "lens": "lens_mode",
+    "anode_library": "anode_material",
+    "start": "energy_start",
+    "end": "energy_end",
+    "centre": "energy_centre",
+    "width": "energy_width",
+    "x-ray_power": "x_ray_power",
+}
+
+KEYS_WITH_UNITS: List[str] = [
+    "filament_current",
+    "filament_bias",
+    "charge_balance",
+    "emission_current",
+    "resolution",
+    "energy_start",
+    "energy_end",
+    "energy_centre",
+    "energy_width",
+    "step_size",
+    "dwell_time",
+    "sweep_time",
+    "x_ray_power",
+]
+
+UNIT_MISSING: Dict[str, str] = {
+    "sample_tilt": "degree",
+}
+
+
+class KratosParser:
+    """
+    A parser for reading in data from Kratos spectrometers.
+    """
+
+    def __init__(self):
+        """
+        Construct the parser.
+
+        """
+        self.raw_data: str = ""
+        self.spectra: List[Dict[str, Any]] = []
+
+        self.metadata = KratosMetadata()
+
+    def parse_file(self, file: Union[str, Path], **kwargs):
+        """
+        TODO: parse actual data, not just metadata!
+
+        Parse the data file into a list of dictionaries.
+
+        Parsed data is stored in the attribute 'self.data'.
+        Each dictionary in the data list is a grouping of related
+        attributes. The dictionaries are later re-structured into a
+        nested dictionary that more closely resembles the domain logic.
+
+        Parameters
+        ----------
+        file : str
+            XPS data filepath.
+
+        Returns
+        -------
+        list
+            Flat list of dictionaries containing one spectrum each.
+
+        """
+        # TODO: write parser for actual Kratos data file
+        # header, data = self._separate_header_and_data()
+        # self.parse_header_into_metadata(header)
+
+        return self.spectra
+
+    def parse_header_into_metadata(self, header: List[str]):
+        """
+        Parse header into KratosMetadata dataclass.
+
+        Parameters
+        ----------
+        header : str
+            Header data for one spectrum as a String.
```

Comment:
Type of parameter in function declaration and function doc is not the same.

Reply:
changed, added more typing for this class